src/EditorFeatures/CSharpTest/Completion/CompletionProviders/TypeImportCompletionProviderTests.cs

```diff
@@ -0,0 +1,564 @@
+        {
+        }
+
+        internal override CompletionProvider CreateCompletionProvider()
+        {
+            return new TypeImportCompletionProvider();
+        }
+
+        private bool? ShowImportCompletionItemsOptionValue { get; set; } = true;
+
+        protected override void SetWorkspaceOptions(TestWorkspace workspace)
+        {
+            workspace.Options = workspace.Options
+                .WithChangedOption(CompletionOptions.ShowImportCompletionItems, LanguageNames.CSharp, ShowImportCompletionItemsOptionValue);
+        }
+
+        #region "Option tests"
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task OptionSetToNull()
+        {
+            ShowImportCompletionItemsOptionValue = null;
+            var markup = @"
+class Bar
+{
+     $$
+}";
+
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task OptionSetToFalse()
+        {
+            ShowImportCompletionItemsOptionValue = false;
+            var markup = @"
+class Bar
+{
+     $$
+}";
+
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task OptionSetToTrue()
+        {
+            ShowImportCompletionItemsOptionValue = true;
+            var markup = @"
+class Bar
+{
+     $$
+}";
+
+            await VerifyAnyItemExistsAsync(markup);
+        }
+
+        #endregion
+
+        #region "CompletionItem tests"
+
+        [InlineData("class", (int)Glyph.ClassPublic)]
+        [InlineData("struct", (int)Glyph.StructurePublic)]
+        [InlineData("enum", (int)Glyph.EnumPublic)]
+        [InlineData("interface", (int)Glyph.InterfacePublic)]
+        [Theory, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task Show_TopLevel_NoImport_InProject(string typeKind, int glyph)
+        {
+            var file1 = $@"
+namespace Foo
+{{
+    public {typeKind} Bar
+    {{}}
+}}";
+            var file2 = @"
+namespace Baz
+{
+    class Bat
+    {
+         $$
+    }
+}";
+            await VerifyItemExistsAsync(
+                CreateMarkupForSingleProject(file2, file1, LanguageNames.CSharp),
+                "Bar",
+                glyph: glyph,
+                inlineDescription: "Foo");
+        }
+
+        [InlineData("class")]
+        [InlineData("struct")]
+        [InlineData("enum")]
+        [InlineData("interface")]
+        [Theory, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task DoNotShow_TopLevel_SmaeNamespace_InProject(string typeKind)
+        {
+            var file1 = $@"
+namespace Foo
+{{
+    public {typeKind} Bar
+    {{}}
+}}";
+            var file2 = @"
+namespace Foo
+{
+    class Bat
+    {
+         $$
+    }
+}";
+            await VerifyItemIsAbsentAsync(
```

Comment:
I guess absent in this particular completion provider? regular symbol completion provider will provide this I guess. so, is the engine know how to handle duplicated items from multiple providers?

Reply:
what happen if there is Bar from regular symbol completion provider (same namespace) and one from unimported one from (this completion provider?)

Reply:
- You can enable just one provider for testing.
- No, it seems VS completion can't handle multiple items with same display name but different suffix. This needs to be fixed in editor. Meanwhile, I will see if there's something I can do to mitigate the issue (maybe attach an invisible char at the end?)

Reply:
I made a hack to not hide items from symbol provider, will talk to editor team for a proper solution